### DISCUSSION TITLE
Change of the multiplication order for covariance calculation

### DISCUSSION THREAD

**Discussion by abhijeetdhakane** [deeptime-ml/deeptime#253] [Q&A]
Hello,

I have a question about, the following code from:

https://github.com/deeptime-ml/deeptime/blob/8dc85b3e40e6b16f78212ac9aa412532ff138ce1/deeptime/decomposition/deep/_vampnet.py

    
    # Calculate the cross-covariance
    y_t = y.transpose(0, 1)
    x_t = x.transpose(0, 1)

    cov_01 = 1 / (batch_size - 1) * torch.matmul(x_t, y)
    # Calculate the auto-correlations
    cov_00 = 1 / (batch_size - 1) * torch.matmul(x_t, x)
    cov_11 = 1 / (batch_size - 1) * torch.matmul(y_t, y)

What  if I change it into

    
    # Calculate the cross-covariance

    y_t = y.transpose(0, 1)
    x_t = x.transpose(0, 1)
    cov_01 = 1 / (batch_size - 1) * torch.matmul(x, y_t)
    # Calculate the auto-correlations
    cov_00 = 1 / (batch_size - 1) * torch.matmul(x, x_t)
    cov_11 = 1 / (batch_size - 1) * torch.matmul(y, y_t)`

Does this still work? (Well it will work, but is it right?)

**Comment by clonker** [comment]
Hey, if we think about `x \in R^{T, n_features}`, then `cov_00` describes the covariances between features. Your change modifies this meaning entirely, I do not think it is right in this context. 🙂

**Reply by abhijeetdhakane** [reply]
You're right!
